### PR TITLE
ol docs

### PR DESCRIPTION
**Pull request by will-moore**
Updated the docs/README.md and the classes diagram to include more details of the OpenLayers viewer.

You can view the page at https://github.com/will-moore/omero-iviewer/tree/ol_docs/docs (shows old diagram). See the new diagram at https://raw.githubusercontent.com/will-moore/omero-iviewer/ol_docs/docs/iviewer_classes.png

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Empty PR description. Please add a short summary of the PR scope and some testing instructions.

File: docs/README.md [hunk 1]
```diff
@@ -93,14 +93,122 @@ being the entry point.
...
+    // source/Regions.js
+
+    setModes(modes) {
+        // if DRAW in modes
+        this.draw_ = new Draw(oldModes, this);
+    }
+
+The viewer/interaction/Draw calls wraps an OpenLayers Draw instance instead of
+extending it. A new ``OpenLayers Draw`` instance is created with a function that
+generates e.g a new viewer/geom/Rectangle:
```
Comment (jburel) [inline docs/README.md]: e.g.

File: docs/iviewer_classes.xml [hunk 2]
```diff
@@ -1 +1 @@
...
...Q1zzxS2e+S0Yig6l4pXAiY/ZKNLGFvC/X6Ih+ZC/cN0E+x+U4rKBLiXzTrsgnLWXSAvm+bCBJ9vsMXmCcvA06WtMKHXVXJKNud0RHqR/VFfOKb/N6NGrnl7eaYizgWYL6vRDnXAmWKUF4Zzu/yCEuph3vqrI1kC9vOGXpABJqJIqIjwH2iZG/n6wIMweCt5dd2gGlm24X00LK6Z4QobhMTu1DMiTvOw3ngu54X8wsT6BCwzBo6HOvGun7eW/OuVWTwHFKQshI5v86GCKvtUPqI4KuHFD9BgAtvT7Qbbjn9de1Y3JKTC6AaPFJUk29hfxrGN3lysq5UfQS3bIaqZLdVfK6uhsQ7fc/PFEhQjwQEYrXIBR9EFjGYz4jZVhROMmHavIdLlKy10vhkMgfc9EGsKUrA/gyDGA+BCqzgOUVQNpYYytnLdGDVy46PKJVRTUf6xD43xWGxwafhPjzv+++/K5Uo9ZVo1zY7VWNrIb7dHSmGumaCBk/Bk/3BVYi4nrXbnBbVv9WYaRvjDjTwTEii8spT0r7pGby+xCpu/KkeNJSxCR6mlytyV/F/X1DomC+wdjfe5X5rb0vmWjFema2U9l6lnd05Kjq1oLms6AOWcR8VtOx/U29JhlQzCW05U87YVkGZ0F7dvVGHVvQuibKaaKgCXx6GSaritleqsnqNC5ayaemtRi8eXMr8s6bKuokQu8eUz7r9cDSoLF5Zn9dO2ot0dECnn9OfxJdUu0xvPczoV4xxFE4CYiYBgtiwkwyu/xBqeWtq+V7OWp/ekPfVrnKb+iJ1k3LvnaW4KCJ4WwiGu5WYAOvEvI3245Ki1Fwj++UfUabVJnr/WNFkvrQs1xg9aNkYJmh9RpFM+TDe5UI0T84JJkQfYNDdOUScNzCRQqiZahgMQQsZCXf+saFWMuR4OIz1h8ySDTb1vAj/lVFe4dAkNljifK6Oi0VBH0i2dd9J3V8DUJYebDK61BOsmOcZPzOLs3zOuwWNk+UM5chiufAvz91A9qP2W+++RX+/+lGCezWBfa+slUDJH44Yu31DREDjfFD8ZKu4H8VUppW/j0wSzauB9xn+ocjupTSYH0KUL4qoPQNlKbJIx2qgMoj2dP0oUtW9ffrktRFl2SKNe9vcfhxG2WxCeVKGAIYRnfTBT6MEbG/dpEvPHqrvLiG+eEf</diagram></mxfile>
```
Comment (jburel) [inline docs/iviewer_classes.xml]: add extra line

**Comment by jburel** [comment]
is there not a way to inject those code snippet?
each time the code is modified, the doc will have to be updated.
Minimally we should add the file the snippet is extracting from

**Comment by jburel** [comment]
Thanks for adding the path to the file

**Pull request merged**